File: webshop/backend/src/main/java/com/example/service/UserService.java
package com.example.service;
import com.example.model.CalcTableEntity;
import com.example.model.UserEntity;
import java.util.List; // Correct import for List




public interface UserService {
    boolean create(UserEntity user);
    void saveUser(UserEntity user);
    boolean update( long id, UserEntity userEntity);
    boolean delete(long id);
    boolean deleteAll();

    UserEntity getById(long id);

    List<UserEntity>getAll();

    List<UserEntity> getOccupiedRecords(String selectedMonth);
  
    List<CalcTableEntity> getCalcTable(String selectedDay, String selectedTimeOfDay);
    
    List<UserEntity> getLowestOccupation();

    List<UserEntity> getHighestOccupation();

    List<UserEntity> getWeekdaysView();

    List<UserEntity> getWeekenddaysView();


}

File: webshop/frontend/webshopapp/src/app/services/user.service.ts
// user.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { UserEntity } from '../model/user.entity.model'; // Make sure to create this model

@Injectable({
  providedIn: 'root',
})
export class UserService {
  private baseUrl = 'http://localhost:8080/UserEntity';

  constructor(private http: HttpClient) { }

  createUser(user: UserEntity): Observable<boolean> {
    return this.http.post<boolean>(`${this.baseUrl}`, user);
  }

  getAllUsers(): Observable<UserEntity[]> {
    return this.http.get<UserEntity[]>(`${this.baseUrl}`);
  }

  deleteUser(userId: number): Observable<boolean> {
    // Construct the correct URL for deletion
    const deleteUrl = `${this.baseUrl}/delete?id=${userId}`;

    // Make the DELETE request
    return this.http.delete<boolean>(deleteUrl);
  }

  getCalcTable(day: string | null, timeofday: string | null): Observable<any> {
    // Construct the base URL
    let url = `${this.baseUrl}/getCalcTable`;
    // Add parameters if they are provided and not null or empty
    if (day && day.trim() !== '') {
      url += `?selectedDay=${day}`;
    }
    if (timeofday && timeofday.trim() !== '') {
      // Use '&' if there's already a parameter
      url += (url.includes('?') ? '&' : '?') + `selectedTimeOfDay=${timeofday}`;
    }
    return this.http.get(url);
  }

  getCurrentStatus(): Observable<any> {
    // Construct the base URL
    let url = `${this.baseUrl}/getCurrentStatus`;
    return this.http.get(url);

  }


}
